[AZINTS-3246] replace the old blob forwarder with the new one (#28536)
DataDog/documentation: content/en/logs/guide/azure-logging-guide.md

Reviewer: Somewhere we should mention the new blob forwarder has automatic updates and will not be a static image

Author: true, let me add that

{{% tab "Blob Storage" %}}

> To get started, click the button below and fill in the form on Azure portal. The Azure resources required to get Azure activity logs and Azure platform logs (including resource logs) into your Datadog account will be deployed for you.

Reviewer 2: ```suggestion
To get started, click **Deploy to Azure** and complete the form in the Azure portal. The required Azure resources for sending Azure activity logs and platform logs (including resource logs) to your Datadog account are deployed automatically.
```

**Note**: Resources can only send logs to storage accounts in the same Azure region.

[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/CustomDeploymentBlade/uri/https%3A%2F%2Fddazurelfo.blob.core.windows.net%2Ftemplates%2Fforwarder.json)

> Once the deployment finishes, a storage account with [Azure Blob Storage][301] services will be deployed, along with a [Container App job][302] and [an environment][303], using the names and specifications chosen.

Reviewer 2: ```suggestion
After the deployment finishes, Azure creates a storage account with [Azure Blob Storage][301], a [Container App job][302], and [an environment][303], using the names and specifications you provided.
```

> After the deployment finishes, a storage account with [Azure Blob Storage][301] services will be deployed, along with a [Container App job][302] and [an environment][303], using the names and specifications chosen.

Reviewer 2: ```suggestion
After the deployment finishes, Azure creates a storage account with [Azure Blob Storage][301] and an [environment][303], and deploys a [Container App job][302], using the names and specifications you provided.
```

> ##### Add diagnostic settings to send logs
> ###### Resource logs

Reviewer 2: Recommend bumping the headers to a H3 and H4 for visibility.

```suggestion
### Add diagnostic settings to send logs
#### Resource logs
```

> For each resource which you would like to [forward resource logs][304], when viewing the resource in the Azure portal, under **Monitoring**, select **Diagnostic settings**. Create a new diagnostic setting, select all the categories you would like to forward, and under **Destination details**, select **Archive to a storage account**, and enter the details of the storage account just created.

Reviewer 2: ```suggestion
To [forward resource logs][304] for a resource, open the resource in the Azure portal. Under **Monitoring**, select **Diagnostic settings**. Create a new diagnostic setting, select the categories you want to forward, and under **Destination details**, select **Archive to a storage account**. Then enter the details of the storage account you just created.
```

> For each resource which you would like to [forward resource logs][304], when viewing the resource in the Azure portal, under **Monitoring**, select **Diagnostic settings**. Create a diagnostic setting, select all the categories you would like to forward, and under **Destination details**, select **Archive to a storage account**, and enter the details of the storage account created by the deployment.

Reviewer 2: ```suggestion
To [forward resource logs][304] for a resources in Azure:
1. In the Azure portal, navigate to the resource.
1. Under **Monitoring**, select **Diagnostic settings**.
1. Create a diagnostic setting and select the categories you want to forward.
1. Under **Destination details**, select **Archive to a storage account**.
1. Enter the details of the storage account created by the deployment.
```

> Alternatively, you can browse all resources with diagnostic setting capability in the [Monitor | Diagnostic settings][305] page of the portal.

Reviewer 2: ```suggestion
Alternatively, you can view all resources that support diagnostic settings on the [**Monitor > Diagnostic settings**][305] page in the Azure portal.
```

Reviewer: Doesn't have to be part of this PR, but we should also recommend [Policies](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/diagnostic-settings-policy)

> ###### Activity logs

Reviewer 2: ```suggestion
#### Activity logs
```

> To forward [Azure Activity log data][306], under the [Monitor | Activity log][307] page of the portal, select **Export Activity Logs**, and create a diagnostic setting pointing to the storage account, the same way as you would for resource logs.

Reviewer 2: ```suggestion
To forward [Azure Activity log data][306]:
1. In the Azure portal, go to the [**Monitor > Activity log**][307] page.
1. Select **Export Activity Logs**.
1. Create a diagnostic setting that points to the same storage account you used for resource logs.
```

{{% tab "Blob Storage" %}}

To get started, click **Deploy to Azure** and complete the form in the Azure portal. The required Azure resources for sending Azure activity logs and platform logs (including resource logs) to your Datadog account are deployed automatically.

> **Note**: Resources can only send logs to storage accounts in the same Azure region.

Reviewer: This note should be closer to the diagnostic settings setup

Author: I thought about that at first, but it matters when they are deploying, since otherwise a customer could deploy and then realize that it isnt in the correct region. Maybe it makes sense to add that info to the diagnostic setting section as well

Reviewer: Event Hub forwarder has this same restriction. In the docs, this is at the end of the deployment section immediately before discussing adding diagnostic settings. Maybe just moving lower so it is still near deployment but also near diagnostic settings.

Reviewer: Could also just swap the places of the two notes

[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/CustomDeploymentBlade/uri/https%3A%2F%2Fddazurelfo.blob.core.windows.net%2Ftemplates%2Fforwarder.json)

> After the deployment finishes, Azure creates a storage account with [Azure Blob Storage][301] and an [environment][303], and deploys a [Container App job][302], using the names and specifications you provided.

Reviewer: Can drop the passive you here

Author: @estherk15 suggested this, but happy to update if we think it's better without

Reviewer: Does seem to be common throughout our docs. My nits around you can be ignored.

> **Note**: The Blob Storage forwarder automatically updates to ensure the latest features and bug fixes.

Reviewer: *nit:  automatically pulls the latest image

Author: Not entirely sure if that's more or less clear here, since customers may not always be familiar with container apps

Reviewer: I'd be fine waiting for customer feedback. I could see customers wanting to know about the update mechanism and permissions involved with the current phrasing.

### Add diagnostic settings to send logs
#### Resource logs

> To [forward resource logs][304] for a resources in Azure:

Reviewer: for a resource

Reviewer 2: ```suggestion
To [forward resource logs][304] for a resource in Azure:
```

1. In the Azure portal, navigate to the resource.
1. Under **Monitoring**, select **Diagnostic settings**.

> 1. [Create a diagnostic setting][308] and select the categories you want to forward.

Reviewer: *nit: select the categories to forward

1. Under **Destination details**, select **Archive to a storage account**.
1. Enter the details of the storage account created by the deployment.

> Alternatively, you can view all resources that support diagnostic settings on the [**Monitor > Diagnostic settings**][305] page in the Azure portal.

Reviewer: *nit Alternatively, view all resources...

#### Activity logs

To [forward Activity logs][306]:
1. In the Azure portal, go to the [**Monitor > Activity log**][307] page.
1. Select **Export Activity Logs**.

> 1. [Create a diagnostic setting][308] and select the categories you want to forward.

Reviewer: *nit: and select categories to forward

1. Under **Destination details**, select **Archive to a storage account**.

> 1. Enter the details of the storage account created by the deployment.

Reviewer: *nit similar to above, we could provide some information on how to find the storage account

### PCI compliance

<div class="alert alert-warning">

> PCI DSS compliance for APM and Log Management is only available for Datadog organizations in the <a href="/getting_started/site/">US1 site</a> using the Event Hub forwarder.

Reviewer: We don't define the other forwarder as an event hub forwarder here. Could this do something like `using the <a href="/?tab=automatedinstallation#setup">Automatic Installation</a> or <a href="?tab=manualinstallation#setup">Manual Installation</a> option.`?

Author: yeah 😔 very annoying that we don't specify that it's event hub based
